packages/core/test/financial-templates/TokenFactoryExclusiveMinter.js

```diff
@@ -0,0 +1,122 @@
+const { didContractThrow } = require("@uma/common");
+
+// Tested Contract
+const TokenFactory = artifacts.require("TokenFactoryExclusiveMinter");
+
+// Helper contracts
+const Token = artifacts.require("SyntheticToken");
+
+const { toWei, toBN } = web3.utils;
+
+contract("TokenFactory", function(accounts) {
+  const contractDeployer = accounts[0];
+  const tokenCreator = accounts[1];
+  const rando = accounts[2];
+
+  let tokenFactory;
+
+  const tokenDetails = {
+    name: "UMA Token",
+    symbol: "UMA",
+    decimals: "18"
+  };
+
+  before(async () => {
+    tokenFactory = await TokenFactory.deployed();
+  });
+  it("Can create new tokens and transfers roles successfully", async () => {
+    const tokenAddress = await tokenFactory.createToken.call(
+      tokenDetails.name,
+      tokenDetails.symbol,
+      tokenDetails.decimals,
+      {
+        from: tokenCreator
+      }
+    );
+    await tokenFactory.createToken(tokenDetails.name, tokenDetails.symbol, tokenDetails.decimals, {
+      from: tokenCreator
+    });
+    const token = await Token.at(tokenAddress);
+
+    // Creator should be only minter
+    assert.isFalse(await token.isMinter(contractDeployer));
+    assert.isTrue(await token.isMinter(tokenCreator));
+
+    // Creator should be only burner
+    assert.isFalse(await token.isBurner(contractDeployer));
+    assert.isTrue(await token.isBurner(tokenCreator));
+
+    // Contract deployer should no longer be capable of adding new roles
+    assert(await didContractThrow(token.resetMinter(rando, { from: contractDeployer })));
```

Comment:
Exact same test as TokenFactory but this line tests for `resetMinter` instead of `addMinter`